src/main/java/org/apache/sysds/api/DMLScript.java

```diff
@@ -277,6 +280,7 @@ public static boolean executeScript( Configuration conf, String[] args )
 			//reset runtime platform and visualize flag
 			setGlobalExecMode(oldrtplatform);
 			EXPLAIN = oldexplain;
+			DMLScript.FED_WORKER_PORTS.clear();
```

Comment:
I would like if this was not a variable in DMLScript, since it would not be available when executing in python.

maybe put it in: 
src/main/java/org/apache/sysds/runtime/controlprogram/context/ExecutionContext.java
or 
not have it.

Maybe the best design is to add another argument to rand, that allows you to specify a federated range of worker addresses.